src/multi/bmp180.cpp

```diff
@@ -31,84 +35,76 @@ auto uncompensated_output_data_register()
   return std::to_array<hal::byte>({ 0xF6 });
 }
 
-auto soft_reset_register()
+// Helper function that creates an array containing the address and message for
+// sensor reset.
+auto soft_reset_config()
 {
-  return std::to_array<hal::byte>({ 0xE0 });
+  hal::byte reset_address = 0xE0;
+  hal::byte reset_setting = 0xB6;
+  return std::to_array<hal::byte>({ reset_address, reset_setting });
 }
 
-// struct to hold the computation variables
-struct computation_variables
+// Helper function to spin until data conversion is complete.
+void wait_for_conversion(hal::i2c* p_i2c, hal::byte p_address)
 {
-  std::int32_t x1, x2, x3;
-  std::int32_t b3, b5, b6;
-  std::uint32_t b4, b7;
-  std::int32_t temperature, pressure;
-};
+  bool conversion_finished = 0;
+  while (conversion_finished == 0) {
+    std::array<hal::byte, 1> buffer;
+    hal::write_then_read(*p_i2c,
+                         p_address,
+                         control_measurement_register(),
+                         buffer,
+                         hal::never_timeout());
+    constexpr auto conversion_status_bit = hal::bit_mask::from(5);
+    conversion_finished = !(hal::bit_extract<conversion_status_bit>(buffer[0]));
+  }
+}
```

Comment:
Maybe a better name would be `conversion_busy`?

```suggestion
  bool conversion_busy = true;
  while (conversion_busy) {
    auto status = hal::write_then_read<1>(*p_i2c,
                         p_address,
                         control_measurement_register(),
                         buffer,
                         hal::never_timeout())[0];
    constexpr auto conversion_status_bit = hal::bit_mask::from(5);
    conversion_busy = hal::bit_extract<conversion_status_bit>(status);
  }
}
```

Use `true` and `false` for bools. Using integers for booleans can be confusing.
Also, `hal::write_then_read<1>` has a means to return an array, so you can remove the std::array buffer.